### PR TITLE
fix(emulator): Hosting emulator should not connect to 0.0.0.0 for emulators

### PR DESCRIPTION
**Pull request by lisp719**
### Description

Fix for (#3121): Fixed to connect to 127.0.0.1 when host is set to 0.0.0.0 in firebase.json

### Scenarios Tested

**public/index.html**

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>Welcome to Firebase Hosting</title>

    <script defer src="/__/firebase/8.2.7/firebase-app.js"></script>
    <script defer src="/__/firebase/8.2.7/firebase-firestore.js"></script>
    <script defer src="/__/firebase/init.js?useEmulator=true"></script>
  </head>
  <body>
    <script>
      document.addEventListener("DOMContentLoaded", function () {
        firebase
          .firestore()
          .doc("/foo/bar")
          .get()
          .then((doc) => {
            console.log(doc.exists);
          });
      });
    </script>
  </body>
</html>
```

**firebase.json**

```json
{
  "firestore": {
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  },
  "hosting": {
    "public": "public",
    "ignore": ["firebase.json", "**/.*", "**/node_modules/**"]
  },
  "emulators": {
    "firestore": {
      "host": "0.0.0.0",
      "port": 8080
    },
    "hosting": {
      "host": "0.0.0.0",
      "port": 5000
    },
    "ui": {
      "enabled": false
    }
  }
}
```

### CODE REVIEW

**Comment by samtstern** [comment]
@lisp719 thank you for this PR! Assigning to @yuchenshi for review.

**Comment by lisp719** [comment]
@googlebot I signed it!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 - Fixes issue where proxied requests to dynamic content through the Hosting emulator would return unexpected `location` headers. (#3097)
 - Fixes issue where optional extension parameters could not be omitted. (#3126)
+- Fixes issue where hosting emulator should not connect to 0.0.0.0 for emulators. (#3121)
```
Comment (yuchenshi) [inline CHANGELOG.md:3]:
```suggestion
- Fixes issue where hosting emulator would connect to 0.0.0.0 for emulators. (#3121)
```

**Review by yuchenshi — APPROVED** [review]

**Comment by yuchenshi** [comment]
Thank you! Merging in this soon.

**Pull request merged**